src/main/connection.cpp

```diff
@@ -26,6 +26,10 @@ Connection::Connection(DatabaseInstance &database) : context(make_shared<ClientC
 Connection::Connection(DuckDB &database) : Connection(*database.instance) {
 }
 
+Connection::~Connection() {
+	ConnectionManager::Get(*database).RemoveConnection(*context);
+}
+
 string Connection::GetProfilingInformation(ProfilerPrintFormat format) {
```

Comment:
Could you add a test case where we open and close a bunch of new connections many times?

Reply:
I've created a test to cover this, let me know what you think